Dockerfile

```diff
@@ -93,19 +93,63 @@ RUN pip install -e . --no-deps
 ###################
 ## Frontend Base ##
 ###################
-FROM node:16-slim as frontend
+FROM node:16-alpine as frontend
+
+RUN apk add --no-cache libc6-compat
+# Build the frontend clients
+WORKDIR /fides/clients
+COPY clients/package.json ./
+COPY clients/package-lock.json ./
+COPY clients/fides-consent/package.json ./fides-consent/package.json
+COPY clients/admin-ui/package.json ./admin-ui/package.json
+COPY clients/privacy-center/package.json ./privacy-center/package.json
 
-# Build the admin-ui frontend
-WORKDIR /fides/clients/admin-ui
-COPY clients/admin-ui/package.json clients/admin-ui/package-lock.json ./
 RUN npm install
-COPY clients/admin-ui/ .
+
+COPY clients/ .
 
 ####################
 ## Built frontend ##
 ####################
 FROM frontend as built_frontend
-RUN npm run export
+
+# Builds and exports admin-ui
+RUN npm run export-ui
+# Builds privacy-center
+RUN npm run build-pc
```

Comment:
another docker optimization thing which we might not need right now, but it might be worth splitting this one into two stages, one for admin ui and one for privacy center. building UI is relatively slow, and we never use _both_ the built admin ui and the built privacy center in the same subsequent stage 

so maybe something like

```Dockerfile
####################
##  Built admin UI ##
####################
FROM frontend as built_admin_ui

# Builds and exports admin-ui
RUN npm run export-ui

####################
## Built privacy center ##
####################
FROM frontend as built_privacy_center
# Builds privacy-center
RUN npm run build-pc

#############################
## Production PC ##
#############################
FROM node:16-alpine as prod_pc
...
COPY --from=built_privacy_center /fides/clients /fides/clients

...

#############################
## Production Application ##
#############################
FROM backend as prod

# Copy frontend build over
COPY --from=built_admin_ui /fides/clients/admin-ui/out/ /fides/src/fides/ui-build/static/admin
```

Reply:
> we never use both the built admin ui and the built privacy center in the same subsequent stage

We don't actually build from dockerfile as part of dev (e.g. `nox -s dev -- pc` stops at the `frontend` target in `Dockerfile`). For this command, the build step happens as a Turbo dependency of when the `docker.compose.yml` calls `npm run dev-pc`.

The only time we run the `built_frontend` stage is for the prod and test (`nox -s "fides_env(test)"`) environments, in which we _do_ subsequently use both the built admin ui and build privacy center for the prod builds (even though those builds are technically separate images).

Reply:
I think that separation makes good sense - and if we keep moving in this direction, we can overall refactor the `build` command to use distinct application names like `build(privacy-center)`, `build(admin-ui)`, `build(fides`), etc.

We did this for the `fidesplus` nox setup and doing the same thing here would be make sense. That can be a follow-up though, no need to expand scope too much here 😄

Reply:
To resolve this, I'll create a ticket for Docker build optimization